### DISCUSSION TITLE
SPI for display use different pins as SPI for touch support (e.g. ESP32-Cheap-Yellow-Display)

### DISCUSSION THREAD

**Discussion by marcelrv** [Bodmer/TFT_eSPI#3123] [Q&A - Touch]
1. A description of the problem and the conditions that cause it to occur
When using the library for the CYD (ESP32-Cheap-Yellow-Display) like the ESP32-2432S028R the issue arrises that the SPI for the display is different from the SPI of the touch.
So when defining the touch like this:
```
#define TFT_MOSI 32
#define TFT_SCLK 25
#define TFT_MISO 39
#define TOUCH_IRQ 36
#define TOUCH_CS 33
```
the touch works well, but the display does not work.

defining the right pins for the display
```
#define TFT_BL 21 // LED back-light control pin
#define TFT_BACKLIGHT_ON HIGH // Level to turn ON back-light (HIGH or LOW)
#define TFT_MISO 12
#define TFT_MOSI 13 // In some display driver board, it might be written as "SDA" and so on.
#define TFT_SCLK 14
#define TFT_CS 15 // Chip select control pin
#define TFT_DC 2 // Data Command control pin
#define TFT_RST -1
#define TOUCH_IRQ 36
#define TOUCH_CS 33
```
ensures the display works well, but the touch no longer works.

2. IDE (e.g. Arduino or PlatformIO)
Arduino 

3. TFT_eSPI library version (try the latest, the problem may have been resolved!) from the Manage Libraries... menu
Latest (2.5.34)

5. Board package version (e.g. 2.0.3) available from the Boards Manager... menu

6. Procesor, e.g RP2040, ESP32 S3 etc
ESP32 https://nl.aliexpress.com/item/1005004961285750.html

7. TFT driver (e.g. ILI9341), a link to the vendors product web page is useful too.
ILI9341

8. Interface type (SPI or parallel)
SPI

Is there a way currently to have this configured / supported by the library?

**Comment by Bodmer** [comment]
The library does not support that however you can use an additional touch library, for example this one:
https://github.com/achillhasler/TFT_eTouch

**Reply by marcelrv** [reply]
thanks, will try that.

**Reply by paelzer** [reply]
@marcelrv 

Hello hello, did you manage to get TFT_EWIDGETS working with the library mentioned by Bodmer? If so, would be great if you could share an example how to set it up. I have a hard to understand the documentation containing in the library. 

Have a nice evening!! 
Boris

**Reply by VaAndCob** [reply]
Hi, I have a mod version of Touch.h and Touch.cpp that make CYD  resistive touch work. (the mod is in TouchRaw/TouchRawZ function. ) this is not my own modified sketch but got it from somewhere for  a year ago. (sorry that I can't remember where I got it)
Here[https://github.com/VaAndCob/ESP32-OBD2-Gauge/tree/main/sketch/library%20modified](url)

**Reply by ggaljoen** [reply]
Made a solution for this board in https://github.com/Bodmer/TFT_eSPI/pull/3186
